File: MenuBack/MenuBack/src/main/java/org/example/menuback/modelo/Menu.java
package org.example.menuback.modelo;


import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.math.BigDecimal;


@Builder
@NoArgsConstructor
@AllArgsConstructor
@Data
@Entity


@Table(name = "menu")
public class Menu {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "id_menu")
    private Long idMenu;

    @Column(name = "nombre", nullable = false)
    private String nombre;

    @Column(name = "descripcion", nullable = false)
    private String descripcion;

    @Column(name = "precio", nullable = false)
    private BigDecimal precio;
}

File: CrudMenu/app/src/main/java/pe/edu/upeu/crudmenu/modelo/Menu.kt
package pe.edu.upeu.crudmenu.modelo

data class MenuDto(
    var idMenu: Long,
    var nombre: String,
    var descripcion: String,
    var precio: Double
)

data class MenuResp(
    val idMenu: Long,
    val nombre: String,
    val descripcion: String,
    val precio: Double
)

fun MenuResp.toDto(): MenuDto {
    return MenuDto(
        idMenu    = this.idMenu,
        nombre    = this.nombre,
        descripcion = this.descripcion,
        precio    = this.precio
    )
}